### PR TITLE
changes the slice bytes transformer to graph stage #6773

### LINKED ISSUE #6773: regression: compilation failure against current Akka master
this came up in the 2.12 community build at https://scala-ci.typesafe.com/job/scala-2.12.x-integrate-community-build/1030/consoleFull:

```
[play2-core] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.7-RC1/project-builds/play2-core-3262e12350c4c6b58f4a66706e3729e6b5d2c9ef/framework/src/play/src/main/scala/play/api/mvc/RangeResult.scala:418: not found: type StatefulStage
[play2-core] [error]     val transformer = new StatefulStage[ByteString, ByteString] {
[play2-core] [error]                           ^
```

it's partly guesswork, but I think this might have to do with https://github.com/akka/akka/pull/21919, cc @ktoso. I opened a similar ticket in the akka-http repo, https://github.com/akka/akka-http/issues/606

for now, to resolve the problem in the community build I have frozen Akka at an earlier commit.

### PR DESCRIPTION
**Pull request by schmitch**
## Fixes

Fixes #6773 

Note to myself: don't code at night.

### CODE REVIEW

File: framework/src/play/src/main/scala/play/api/mvc/RangeResult.scala [hunk 1]
```diff
@@ -415,36 +421,39 @@ object RangeResult {
 
   // See https://github.com/akka/akka/blob/v2.4.2/akka-http-core/src/main/scala/akka/http/impl/util/StreamUtils.scala#L83-L115
```
Comment (marcospereira) [inline framework/src/play/src/main/scala/play/api/mvc/RangeResult.scala]: You can remove this comment or updated it to https://github.com/akka/akka-http/blob/master/akka-http-core/src/main/scala/akka/http/impl/util/StreamUtils.scala#L76

File: framework/src/play/src/main/scala/play/api/mvc/RangeResult.scala [hunk 2]
```diff
@@ -415,36 +421,39 @@ object RangeResult {
...
   private def sliceBytesTransformer(start: Long, length: Option[Long]): Flow[ByteString, ByteString, NotUsed] = {
-    val transformer = new StatefulStage[ByteString, ByteString] {
-
-      def skipping = new State {
-        var toSkip = start
-
-        override def onPush(element: ByteString, ctx: Context[ByteString]): SyncDirective =
-          if (element.length < toSkip) {
-            // keep skipping
+    val transformer = new GraphStage[FlowShape[ByteString, ByteString]] {
```
Comment (marcospereira) [inline framework/src/play/src/main/scala/play/api/mvc/RangeResult.scala]: akka-http-core (linked above) is using a `SimpleLinearGraphStage` which looks more simple and direct.

Comment (schmitch) [inline framework/src/play/src/main/scala/play/api/mvc/RangeResult.scala]: it's a internal api. not a good idea to use.

Comment (marcospereira) [inline framework/src/play/src/main/scala/play/api/mvc/RangeResult.scala]: Ok. Interesting that this is a Akka internal API but is used by akka-http.

**Pull request merged**